### PR TITLE
refine task assertions, add test for task timeout

### PR DESCRIPTION
**Pull request by guybedford**
This adds a test for a timed out fetch request, that relies on the timeout feature of `async_select`.

This is currently failing on compute due to the recently fixed bug at https://github.com/fastly/ExecuteD/pull/4243, which should then pass once that rollout occurs.

This PR also refines the event loop task assertions and errors per debugging for this error.

### CODE REVIEW

File: integration-tests/js-compute/fixtures/app/src/timers.js [hunk 1]
```diff
@@ -342,6 +342,35 @@ import { routes } from "./routes.js";
...
+        let timedOut = false
+        const first = fetch('https://httpbin.org/delay/1', {
+            backend: 'httpbin'
+        })
+        const second = Promise.race([
+            fetch('https://httpbin.org/delay/1', {
+                backend: 'httpbin'
+            }),
+            new Promise(resolve => setTimeout(resolve, 5)).then(() => {
+                console.error('TIMEOUT');
```
Comment (guybedford) [inline integration-tests/js-compute/fixtures/app/src/timers.js:355]: These logs should be removed once the test is passing!

File: runtime/fastly/host-api/host_api.cpp [hunk 2]
```diff
@@ -127,13 +127,16 @@ size_t api::AsyncTask::select(std::vector<api::AsyncTask *> &tasks) {
   fastly::fastly_host_error err = 0;
 
   while (true) {
-    if (!convert_result(fastly::async_select(handles.data(), handles.size(),
-                                             (soonest_deadline - now) / MILLISECS_IN_NANOSECS,
-                                             &ret),
+    MOZ_ASSERT(soonest_deadline == 0 || soonest_deadline > now);
+    uint32_t timeout = soonest_deadline > 0 ? (soonest_deadline - now) / MILLISECS_IN_NANOSECS : 0;
```
Comment (elliottt) [inline runtime/fastly/host-api/host_api.cpp:131]: Is it worth adding a comment here the `0` means no timeout?

**Review by elliottt — APPROVED** [review]
This looks great!

File: runtime/fastly/host-api/host_api.cpp (hunk 2 above)
Comment (guybedford) [reply runtime/fastly/host-api/host_api.cpp:131]: Sure, how about this:

```suggestion
    // timeout value of 0 means no timeout for async_select
    uint32_t timeout = soonest_deadline > 0 ? (soonest_deadline - now) / MILLISECS_IN_NANOSECS : 0;
```

Comment (elliottt) [reply runtime/fastly/host-api/host_api.cpp:132]: Perfect!

**Pull request merged**